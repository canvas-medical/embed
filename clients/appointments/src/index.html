<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <title>Appointments</title>
    <meta name="viewport" content="width=device-width,initial-scale=1" />
    <style>
      @import url('https://fonts.googleapis.com/css2?family=Roboto:wght@400;700&display=swap');

      body {
        margin: 0;
      }
    </style>
  </head>
  <body>
    <div id="embed-root"></div>
    <script src="./appointments.js" type="text/javascript"></script>
    <script type="text/javascript">
      const callbacks = {
        onClick: (e, config) => {
          if (e && e.target) {
            // Prescribed method for getting the element clicked
            // const closestTarget = e.target.closest('[data-analytics-id]')
            // console.log(
            //   closestTarget.dataset,
            //   'e.target.dataset',
            //   config,
            //   'config'
            // )
          }
        },
        onChange: (e, config) => {
          if (e && e.target) {
            // const closestTarget = e.target.closest('[data-analytics-id]')
            // console.log(
            //   closestTarget.dataset,
            //   'e.target.dataset',
            //   config,
            //   'config'
            // )
          }
        },
        onError: (error, msg) => {
          // console.error('error', error, msg)
        },
        onLoad: () => {},
      }

      const config = {
        apiKey: 'API_KEY',
        patientId: 'PATIENT_ID',
        providerIds: ['PROVIDER_ID'],
      }

      const apiBaseUrl = 'http://localhost:3000'

      fetch(
        `${apiBaseUrl}/Auth?` +
          new URLSearchParams({
            key: config.apiKey,
            patient: config.patientId,
          })
      )
        .then(res => res.json())
        .then(data => {
          const patientKey = data.patient_key

          Appointments.init({
            api: apiBaseUrl,
            bailoutURL: 'https://canvasmedical.com',
            locationId: '1',
            patientId: config.patientId,
            patientKey: patientKey,
            rootId: 'embed-root',
            callbacks,
            // brandColor: '',
            // accentColor: '',
            // fontFamily: 'Arial, serif',
          })
        })
    </script>
  </body>
</html>
