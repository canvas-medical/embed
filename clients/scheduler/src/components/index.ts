export * from './confirmation'
export * from './date-select'
export * from './time-slot-select'
