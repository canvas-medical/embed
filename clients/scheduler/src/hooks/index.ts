export * from './app-context'
export * from './use-previous-value'
