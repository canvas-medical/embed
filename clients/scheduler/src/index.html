<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <title>Scheduler</title>
    <meta name="viewport" content="width=device-width,initial-scale=1" />
    <style>
      body {
        margin: 0;
        overflow: hidden;
      }
    </style>
    <link rel="preconnect" href="https://fonts.googleapis.com" />
    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin />
    <link
      href="https://fonts.googleapis.com/css2?family=Roboto:wght@400;700&display=swap"
      rel="stylesheet"
    />
  </head>
  <body>
    <div id="embed-root"></div>
    <script src="./scheduler.js" type="text/javascript"></script>
    <script type="text/javascript">
      const callbacks = {
        onClick: (e, config) => {
          if (e && e.target) {
            // Prescribed method for getting the element clicked
            // const closestTarget = e.target.closest('[data-analytics-id]')
            // console.log(
            //   closestTarget.dataset,
            //   'e.target.dataset',
            //   config,
            //   'config'
            // )
          }
        },
        onChange: (e, config) => {
          if (e && e.target) {
            // const closestTarget = e.target.closest('[data-analytics-id]')
            // console.log(
            //   closestTarget.dataset,
            //   'e.target.dataset',
            //   config,
            //   'config'
            // )
          }
        },
        onError: (error, msg) => {
          // console.error('error', error, msg)
        },
        onLoad: () => {},
        onDateChange: config => {
          // console.log(config, 'config')
        },
      }

      const config = {
        apiKey: 'API_KEY',
        patientId: 'PATIENT_ID',
        providerIds: ['PROVIDER_ID'],
      }

      // This is the URL of the running proxy container
      const apiBaseUrl = 'http://localhost:3000'

      fetch(
        `${apiBaseUrl}/Auth?` +
          new URLSearchParams({
            key: config.apiKey,
            patient: config.patientId,
          })
      )
        .then(res => res.json())
        .then(data => {
          const patientKey = data.patient_key

          Scheduler.init({
            api: apiBaseUrl,
            appointmentCoding: {
              code: 'CODE',
              system: 'http://snomed.info/sct',
            },
            bailoutURL: 'https://canvasmedical.com',
            callbacks: callbacks,
            daysToFetch: '30',
            duration: '60',
            locationId: '1',
            patientId: config.patientId,
            patientKey: patientKey,
            providerIds: config.providerIds,
            reason: 'high fever, cough, leeches',
            returnURL: 'https://canvasmedical.com',
            rootId: 'embed-root',
            // brandColor: '',
            // accentColor: '',
            // customFont: 'Nunito',
          })
        })
        .catch(e => {
          console.error(e)
        })
    </script>
  </body>
</html>
