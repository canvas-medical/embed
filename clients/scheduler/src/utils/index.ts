export * from './functions'
export * from './dates'
export * from './types'
