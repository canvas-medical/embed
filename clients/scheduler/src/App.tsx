import { h } from 'preact'
import { AppContainer, Body, Error, Header } from '@canvas/embed-common'
import { useAppContext } from './hooks'
import { Confirmation, DateSelect, TimeSlotSelect } from './components'

export const App = () => {
  const { bailoutURL, colors, screen, error } = useAppContext()
  return (
    <AppContainer>
      <Header
        bailoutURL={bailoutURL}
        colors={colors}
        title="Schedule an Appointment"
      />
      {error && error.length ? (
        <Error errorMessages={error} />
      ) : screen !== 'CONFIRM' ? (
        <Body>
          <DateSelect />
          <TimeSlotSelect />
        </Body>
      ) : (
        <Body>
          <Confirmation />
        </Body>
      )}
    </AppContainer>
  )
}
