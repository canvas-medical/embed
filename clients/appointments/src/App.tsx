import { h } from 'preact'
import { AppContainer, Header } from '@canvas-medical/embed-common'
import { IAppProps } from './utils'
import { AppointmentsView } from './components'

export const App = ({
  api,
  bailoutURL,
  locationId,
  patientId,
  patientKey,
  providerIds,
  shadowRoot,
  colors,
  customFont,
}: IAppProps) => {
  return (
    <AppContainer customFont={customFont}>
      <Header
        bailoutURL={bailoutURL}
        colors={colors}
        title="Your Appointments"
      />
      <AppointmentsView
        api={api}
        bailoutURL={bailoutURL}
        colors={colors}
        locationId={locationId}
        patientId={patientId}
        patientKey={patientKey}
        providerIds={providerIds}
        shadowRoot={shadowRoot}
      />
    </AppContainer>
  )
}
