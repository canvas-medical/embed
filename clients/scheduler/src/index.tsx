import { h, render } from 'preact'
import { StyleSheetManager } from 'styled-components'
import {
  css,
  Error,
  generateColors,
  getAppointmentType,
} from '@canvas/embed-common'
import { ContextWrapper } from './hooks'
import { App } from './App'
import { hasAllValues, iInitializerProps, iSchedulerProps } from './utils'

export const Scheduler = (props: iSchedulerProps) => {
  const {
    api,
    appointmentTypeCode,
    bailoutURL,
    duration,
    locationId,
    patientId,
    patientKey,
    providers,
    reason,
    returnURL,
    brandColor,
    accentColor,
    shadowRoot,
  } = props
  const colors = generateColors(brandColor, accentColor)
  const treatment = getAppointmentType(appointmentTypeCode)
  const allValuesProvided = hasAllValues(props)

  return (
    // Ignoring type mismatch error on target - ShadowRoot is an acceptable type
    // @ts-ignore
    <StyleSheetManager target={shadowRoot}>
      {!allValuesProvided.length ? (
        <ContextWrapper
          values={{
            api,
            bailoutURL,
            duration,
            locationId,
            patientId,
            patientKey,
            providers,
            colors,
            reason,
            returnURL,
            shadowRoot,
            treatment,
            loading: false,
            screen: 'SELECT',
            timeSlot: {
              start: '',
              end: '',
              provider: {
                id: '',
                name: '',
              },
            },
            setTimeSlot: () => {},
            resetTimeSlot: () => {},
            date: new Date(),
            setDate: () => {},
            fetchTimeSlots: () => {},
            fetchScheduledAppointment: () => {},
            createAppointment: () => {},
            cancelAppointment: () => {},
          }}
        >
          <App />
        </ContextWrapper>
      ) : (
        <Error errorMessage={allValuesProvided} />
      )}
    </StyleSheetManager>
  )
}

export const init = ({
  api,
  appointmentTypeCode,
  bailoutURL,
  duration,
  locationId,
  patientId,
  patientKey,
  providers,
  reason,
  returnURL,
  rootId,
  brandColor,
  accentColor,
}: iInitializerProps) => {
  const appRoot = document.querySelector(`#${rootId}`)

  if (!appRoot) {
    console.error('App root could not be found. Check your rootId')
    return null
  }

  appRoot.attachShadow({
    mode: 'open',
  })

  if (!appRoot.shadowRoot) {
    console.error('Shadow root could not be attached')
    return null
  }

  const styleTag = document.createElement('style')
  styleTag.innerHTML = css
  appRoot.shadowRoot.appendChild(styleTag)

  render(
    <Scheduler
      api={api}
      appointmentTypeCode={appointmentTypeCode}
      bailoutURL={bailoutURL}
      duration={duration}
      locationId={locationId}
      patientId={patientId}
      patientKey={patientKey}
      providers={providers}
      reason={reason}
      returnURL={returnURL}
      brandColor={brandColor}
      accentColor={accentColor}
      shadowRoot={appRoot.shadowRoot}
    />,
    appRoot.shadowRoot
  )
}
