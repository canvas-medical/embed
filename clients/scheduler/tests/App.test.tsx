import { h } from 'preact'
import { render } from 'preact-render-to-string'
import { App } from '../src/App'

describe('App', () => {
  it('renders', () => {
    expect(render(<App />)).toMatchSnapshot()
  })
})
