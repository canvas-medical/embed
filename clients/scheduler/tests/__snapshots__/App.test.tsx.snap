// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `"<div class=\\"sc-gsDKAQ iMCeHp\\"><div class=\\"sc-ksdxgE jAELwq\\"><div class=\\"sc-hBUSln eZVHDj\\"><a href class=\\"sc-egiyK gAFTBw\\"><svg height=\\"16\\" viewBox=\\"0 0 10 16\\" width=\\"10\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"m401.034779 24.5989021v10h10\\" fill=\\"none\\" stroke=\\"#fff\\" stroke-width=\\"2\\" transform=\\"matrix(.70710678 .70710678 -.70710678 .70710678 -257.145582 -299.440628)\\"></path></svg></a><div class=\\"sc-fotOHu bQqSAl\\"><h1 class=\\"sc-furwcr dnfnN\\">Schedule an Appointment</h1></div></div></div><div class=\\"sc-dkPtRN cFfYed\\"><div class=\\"sc-hKwDye KNrOE\\"><div class=\\"sc-ieecCq gTdlWo\\"><button aria-label=\\"Scroll Date Back\\" disabled class=\\"sc-dJjYzT cQpbJY\\"><svg height=\\"16\\" viewBox=\\"0 0 10 16\\" width=\\"10\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"m401.034779 24.5989021v10h10\\" fill=\\"none\\" stroke=\\"#fff\\" stroke-width=\\"2\\" transform=\\"matrix(.70710678 .70710678 -.70710678 .70710678 -257.145582 -299.440628)\\"></path></svg></button><button aria-label=\\"Open Date Picker\\" class=\\"sc-hGPBjI pdvRF\\"><svg clip-rule=\\"evenodd\\" fill-rule=\\"evenodd\\" stroke-linejoin=\\"round\\" stroke-miterlimit=\\"2\\" width=\\"20\\" height=\\"21\\" viewBox=\\"0 0 34 36\\" xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"#000\\"><path d=\\"m25.763 0c.744 0 1.341.594 1.354 1.341v.023.834h2.595c2.383 0 4.287 1.931 4.288 4.286l-.001.072v25.086c0 2.364-1.916 4.319-4.254 4.357l-.071.001h-25.349c-2.346 0-4.286-1.931-4.324-4.286l-.001-.072v-25.048c0-2.364 1.917-4.319 4.254-4.358h.071 2.595v-.872c0-.758.602-1.364 1.354-1.364.745 0 1.342.594 1.354 1.341v.023.872h6.055v-.872c0-.758.602-1.364 1.354-1.364.745 0 1.342.594 1.354 1.341v.023.872h6.018v-.872c0-.758.602-1.364 1.354-1.364zm5.566 11.861h-28.621v19.781c0 .901.737 1.653 1.628 1.668h.027 25.349c.893 0 1.64-.743 1.618-1.64l-.001-.028zm-21.062 15.499c.595 0 1.08.481 1.091 1.078v.021 1.175c0 .599-.477 1.087-1.069 1.099h-.022-3.272c-.594 0-1.079-.481-1.09-1.078v-.021-1.175c0-.599.477-1.087 1.069-1.099h.021zm16.775 0c.632 0 1.117.482 1.128 1.078v.021 1.175c0 .599-.478 1.087-1.07 1.098l-.021.001h-3.272c-.594 0-1.079-.481-1.09-1.078v-.021-1.175c0-.599.477-1.087 1.069-1.099h.021zm-8.387 0c.594 0 1.079.481 1.09 1.078v.021 1.175c0 .599-.477 1.087-1.069 1.098l-.021.001h-3.272c-.595 0-1.079-.481-1.091-1.078v-.021-1.175c0-.599.477-1.087 1.069-1.099h.022zm-8.388-6.404c.595 0 1.08.481 1.091 1.078v.021 1.175c0 .599-.477 1.087-1.069 1.098l-.022.001-3.272-.001c-.594 0-1.079-.481-1.09-1.077v-.022-1.174c0-.599.477-1.087 1.069-1.099h.021zm16.775 0c.632 0 1.117.481 1.128 1.078v.021 1.174c0 .599-.478 1.088-1.07 1.099h-.021-3.272c-.594 0-1.079-.481-1.09-1.077v-.022-1.174c0-.599.477-1.087 1.069-1.099h.021zm-8.387 0c.594 0 1.079.481 1.09 1.077v.022 1.174c0 .599-.477 1.088-1.069 1.099h-.021-3.272c-.595 0-1.079-.481-1.091-1.077v-.022-1.174c0-.599.477-1.087 1.069-1.099h.022zm-8.388-6.366c.595 0 1.08.481 1.091 1.077v.022 1.174c0 .599-.477 1.088-1.069 1.099h-.022-3.272c-.594 0-1.079-.481-1.09-1.078v-.021-1.175c0-.599.477-1.087 1.069-1.098h.021zm16.775 0c.632 0 1.117.481 1.127 1.077l.001.021v1.175c0 .599-.478 1.087-1.07 1.099h-.021-3.272c-.595 0-1.079-.481-1.09-1.078l-.001-.021.001-1.175c0-.599.477-1.087 1.069-1.098h.021zm-8.387 0c.594 0 1.079.481 1.09 1.077v.021 1.175c0 .599-.477 1.087-1.069 1.099h-.021-3.272c-.595 0-1.079-.481-1.091-1.078v-.021-1.175c0-.599.477-1.087 1.069-1.098h.022zm-11.697-9.664h-2.595c-.894 0-1.64.706-1.655 1.639v2.606h28.734v-2.577c0-.901-.737-1.652-1.628-1.667l-.027-.001h-2.595v.872c0 .758-.602 1.364-1.354 1.364-.744 0-1.341-.594-1.354-1.34v-.024-.872h-6.055v.872c0 .758-.602 1.364-1.354 1.364-.744 0-1.341-.594-1.354-1.34v-.024-.872h-6.055v.872c0 .758-.602 1.364-1.354 1.364-.744 0-1.342-.594-1.354-1.34v-.024z\\" fill-rule=\\"nonzero\\"></path></svg><div class=\\"sc-hKwDye bKQjvd\\"><h2 class=\\"sc-pVTFL jWPgjN\\">Feb 22, 2022</h2></div></button><button aria-label=\\"Scroll Date Forward\\" class=\\"sc-dJjYzT cQpbJY\\"><svg height=\\"16\\" viewBox=\\"0 0 10 16\\" width=\\"10\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"m862.034779 91.5989021v9.9999999h10\\" fill=\\"none\\" stroke=\\"#fff\\" stroke-width=\\"2\\" transform=\\"matrix(-.70710678 .70710678 .70710678 .70710678 545.815212 -672.793008)\\"></path></svg></button></div><div class=\\"sc-hKwDye gNMoSX\\"><span font-size=\\"0.875rem\\" class=\\"sc-iqseJM ejrlIa\\">Appointment times shown in UTC</span></div></div><div class=\\"sc-bdvvtL biLwMd\\"><div class=\\"sc-hKwDye cQHjzB\\"><div class=\\"sc-fFeiMQ jjjCFF\\"></div></div></div></div></div>"`;
